src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.producer.BulldozerProducer;
+import core.mate.academy.producer.ExcavatorProducer;
+import core.mate.academy.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+
+        if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            machines = new ArrayList<>(excavatorProducer.produce());
+        }
+        if (type == Bulldozer.class) {
+            BulldozerProducer dozerProducer = new BulldozerProducer();
+            machines = new ArrayList<>(dozerProducer.produce());
+        }
+        if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            machines = new ArrayList<>(truckProducer.produce());
+        }
+        return machines;
```

Comment:
lets create class level variables for producers - to avoid creating a new instance each time this method is invoked
remember about OOP principles - use abstract reference type